### DISCUSSION TITLE
[Support] Swagger 是否支持 参数： persistAuthorization？

### DISCUSSION THREAD

**Discussion by pandaApe** [midwayjs/midway#3471] [Q&A]
### Describe the problem（描述问题）

目前在使用 swagger 插件中，swagger 页面一刷新，token 信息就会丢失。Swagger 官方提供了[persistAuthorization](https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/?sbsearch=persistAuthorization)这个参数，来解决这个问题。请问 midway 的 swagger插件中如何来配置来解决这个问题？

### Midway Versions（Midway 版本）

最新的

**Comment by pandaApe** [comment]
@czy88840616  哥，看看这个

**Reply by czy88840616** [reply]
看到啦

**Reply by czy88840616** [reply]
这个不是 swagger 的配置，是 swagger-ui 的。

**Reply by pandaApe** [reply]
@czy88840616 那该怎么设置呢？ 现在页面一刷新，token 信息就会没了，还得重新设置😭

**Reply by czy88840616** [reply]
看起来这个改动 https://github.com/midwayjs/midway/pull/3528 就能支持了，你可以试试当前的 beta 版本，我已经默认加上了 persistAuthorization: true,

**Reply by pandaApe** [reply]
@czy88840616 谢谢哥🙏。放假了没休息，还在 coding，非常感谢

**Reply by pandaApe** [reply]
@czy88840616  上面问题试了最新版本上，我这边还是存在的。
我注意到，这个改动已经再 v3.14.0 中 上线了。 但是我upgrade 到 3.14.0 后，将`config.default.ts` 中的 swagger 配置改为下面的， 还是有问题。 我看了文档，我是属于安装了`swagger-ui-dist` 包的情况。

```
swagger: {
    swaggerPath: '/api/swagger-ui',
    title: 'myproject',
    description: "Api Doc",
    tagSortable: true,
    auth: {
      authType: 'bearer',
    }
  }
```

**Reply by czy88840616** [reply]
对的，给你的测试版本的是走了 `renderSwaggerUIRemote ` 这个方法，正式版本这个配置默认还是走 ui-dist，差异在这里
https://midwayjs.org/docs/extensions/swagger#%E4%BB%8E-unpkg-%E7%AD%89-cdn-%E5%9C%B0%E5%9D%80%E6%B8%B2%E6%9F%93

**Reply by pandaApe** [reply]
@czy88840616  可能我没表达清楚。 我用的正是正式版的最新的版本是v3.14.2。
 `已经默认加上了 persistAuthorization: true` 这个没有起作用。

**Reply by czy88840616** [reply]
所以你配了 renderSwaggerUIRemote 吗？没配的话需要自己加 swagger-initializer.js 文件。

**Reply by pandaApe** [reply]
@czy88840616  好的明白了。 看了文档里的示例`swagger-initializer.js` , 有完整的文件么？ 示例里面导包的信息不完整。

**Reply by czy88840616** [reply]
这个文件的内容就只有这些，https://unpkg.com/browse/swagger-ui-dist@5.11.0/swagger-initializer.js
